### PR TITLE
Ignore small bursts of high freq metrics

### PR DESCRIPTION
**Pull request by malish8632**
This change will allow to ignore small bursts of high frequency metrics. 
Some of existing systems duplicate their metric production and that skews high frequency detector.

+ @ao2017

### CODE REVIEW

**Review by sjoeboo — APPROVED** [review]
👍

File: api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java [hunk 1]
```diff
@@ -123,15 +128,23 @@ public HighFrequencyDetector() {
...
                     (list.get(size - x).getTime().getTime()
                         - list.get(size - x - 1).getTime().getTime()))
         .filter(d -> (d >= 0 && d < minFrequencyMillisAllowed))
-        .min()
-        .orElse(-1);
+        .summaryStatistics();
+
+    int result = -1;
+
+    if (stats.getCount() > 5) {
```
Comment (ao2017) [inline api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java]: What is the rational of using 5 as the threshold?  It is better not to hardcode the threshold. 
Public static final int THRESHOLD = 5;  if(stats.getCount() > THRESHOLD)

Comment (malish8632) [reply api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java]: It's magic number :) - I'll change

Comment (ao2017) [reply api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java]: :), I am curious why 5

Comment (malish8632) [reply api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java]: It is based on preliminary testing across 65 canary instances.

**Review by ao2017 — APPROVED** [review]

**Pull request merged**